scapy/arch/linux.py

```diff
@@ -228,66 +230,89 @@ def get_alias_address(iface_name,  # type: str
 
 def read_routes():
     # type: () -> List[Tuple[int, int, str, str, str, int]]
+    """
+    Read routes from /proc/net/route
+    """
     try:
-        with open("/proc/net/route", "r") as f:
-            entries = [line.split() for line in f]
+        with open("/proc/net/route", "r", errors='replace') as f:
```

Comment:
Interesting, I was aware of the fact that switching from binary mode + `plain_str()` conversion to text mode could potentially introduce decoding errors, but then I thought that (if at all) the interface names would be utf-8.

Using `errors='replace'` is an elegant solution, I didn't know that existed.

Reply:
The question here actually was: is actually possible that /proc/net/route contains non-ascii characters?

Reply:
You can hurt yourself badly for sure. (we don't support that btw, future todo)

```
user@host:~$ sudo ip link add toto type veth peer name 🥲
user@host:~$ cat /proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
    lo:     200       4    0    0    0     0          0         0      200       4    0    0    0     0       0          0
  🥲:         0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
  toto:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
```

Reply:
That convinces me :rofl:

Reply:
But it's still utf8, which is the default for decoding text files on linux. I wouldn't get an exception, would I?

Reply:
Okay so you're actually right, but just for the sake of it let's push it to a theoretical madness. Here's the é character in latin-1 (e.g. default on many Windows devices)
```
user@host:~$ sudo ip link add � type veth peer name a
user@host:~$ # yeah that actually worked
user@host:~$ cat /proc/net/dev
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
    lo:     200       4    0    0    0     0          0         0      200       4    0    0    0     0       0          0
 bond0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
dummy0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
 tunl0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
  sit0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
  eth0: 427563127  265726    0    0    0     0          0      9435  4244917   55919    0    0    0     0       0          0
     a:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
   �:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
```

Reply:
Obviously this is ridiculous and you're right, but let's keep it anyways :p

Reply:
Sure, the `errors='replace'` catch is useful without any doubt.